File: KilitliDolapBulmacasi/KilitliDolapBulmacasi.java
package kilitlidolapbulmacasi;

import java.util.HashMap;

public class KilitliDolapBulmacasi {

    public static void main(String[] args) {
        HashMap dolaplar = dolaplariKapat();
        ogrenciSirasi(dolaplar);
        acik_dolaplar(dolaplar);
    }

    public static HashMap dolaplariKapat() {
        HashMap dolaplar = new HashMap<>();
        for (int i = 1; i < 101; i++) {
            dolaplar.put(i, "kapalı");
        }
        return dolaplar;
    }

    public static void ogrenciSirasi(HashMap dolaplar) {
        for (int i = 1; i < 101; i++) {
            dolaplariAcKapat(i, dolaplar);
        }
    }

    public static void dolaplariAcKapat(int basamak, HashMap dolaplar) {
        for (int i = basamak; i < 101; i += basamak) {
            if (dolaplar.get(i) == "kapalı") {
                dolaplar.replace(i, "açık");
            } else {
                dolaplar.replace(i, "kapalı");
            }
        }
    }

    public static void acik_dolaplar(HashMap dolaplar) {
        for (int i = 1; i < 101; i++) {
            if (dolaplar.get(i) == "açık") {
                System.out.print(i + " ");
            }
        }
    }
}

File: KilitliDolapBulmacasi/KilitliDolapBulmacasi.py
def dolaplari_kapat():

    dolaplar = {}

    for i in range(1, 101):
        dolaplar[i] = 'kapalı'

    return dolaplar


def ogrenci_sirasi(dolaplar):

    for i in range(1, 101):
        dolaplari_ac_kapat(i, dolaplar)


def dolaplari_ac_kapat(basamak, dolaplar):

    for i in range(basamak, 101, basamak):
        if dolaplar[i] == 'kapalı':
            dolaplar[i] = 'açık'
        else:
            dolaplar[i] = 'kapalı'


def acik_dolaplar(dolaplar):

    for i in range(1, 101):
        if dolaplar[i] == 'açık':
            print(i, end=' ')


dolaplar = dolaplari_kapat()
ogrenci_sirasi(dolaplar)
acik_dolaplar(dolaplar)